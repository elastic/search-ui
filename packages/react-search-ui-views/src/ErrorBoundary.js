import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "./view-helpers";

function ErrorBoundary({ children, className, error, ...rest }) {
  if (error) {
    return (
      <div className={appendClassName("sui-search-error", className)} {...rest}>
        {error}
      </div>
    );
  }

  return children;
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
  error: PropTypes.string.isRequired,
  className: PropTypes.string
};

export default ErrorBoundary;
