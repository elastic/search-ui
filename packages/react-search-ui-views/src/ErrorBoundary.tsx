import React from "react";

import { appendClassName } from "./view-helpers";
import type { SearchContextState } from "@elastic/search-ui";
import { BaseContainerProps } from "./types";

export type ErrorBoundaryContainerContext = Pick<SearchContextState, "error">;

export type ErrorBoundaryViewProps = BaseContainerProps &
  ErrorBoundaryContainerContext;

function ErrorBoundary({
  children,
  className,
  error,
  ...rest
}: ErrorBoundaryViewProps & React.HTMLAttributes<HTMLDivElement>): JSX.Element {
  if (error) {
    return (
      <div className={appendClassName("sui-search-error", className)} {...rest}>
        {error}
      </div>
    );
  }

  return children as JSX.Element;
}

export default ErrorBoundary;
