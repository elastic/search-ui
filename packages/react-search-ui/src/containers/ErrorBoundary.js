import PropTypes from "prop-types";
import React, { Component } from "react";
import { withSearch } from "..";
import { ErrorBoundary } from "@elastic/react-search-ui-views";

export class ErrorBoundaryContainer extends Component {
  static propTypes = {
    // Props
    children: PropTypes.node.isRequired,
    view: PropTypes.func,
    // State
    error: PropTypes.string.isRequired
  };

  render() {
    const { view, ...rest } = this.props;

    const View = view || ErrorBoundary;

    return View(rest);
  }
}

export default withSearch(ErrorBoundaryContainer);
