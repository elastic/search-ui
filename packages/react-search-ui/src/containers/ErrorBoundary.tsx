import React, { Component } from "react";
import { withSearch } from "..";
import {
  ErrorBoundary,
  ErrorBoundaryContainerContext,
  ErrorBoundaryViewProps
} from "@elastic/react-search-ui-views";
import { BaseContainerProps } from "../types";

type ErrorBoundaryContainerProps = BaseContainerProps &
  ErrorBoundaryContainerContext & {
    view?: React.ComponentType<ErrorBoundaryViewProps>;
  };

export class ErrorBoundaryContainer extends Component<ErrorBoundaryContainerProps> {
  render() {
    const { children, className, error, view, ...rest } = this.props;

    const View = view || ErrorBoundary;

    const viewProps = {
      className,
      children,
      error,
      ...rest
    };

    return <View {...viewProps} />;
  }
}

export default withSearch<
  ErrorBoundaryContainerProps,
  ErrorBoundaryContainerContext
>(({ error }) => ({ error }))(ErrorBoundaryContainer);
