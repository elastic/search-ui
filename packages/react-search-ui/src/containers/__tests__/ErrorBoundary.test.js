import React from "react";
import { shallow } from "enzyme";
import { ErrorBoundaryContainer } from "../ErrorBoundary";

const params = {
  children: <div>Child</div>,
  error: "I am an error"
};

it("supports a render prop", () => {
  // eslint-disable-next-line react/prop-types
  const render = ({ error }) => {
    return <div>{error}</div>;
  };
  const wrapper = shallow(<ErrorBoundaryContainer {...params} view={render} />);
  expect(wrapper).toMatchSnapshot();
});

it("passes className through to the view", () => {
  let viewProps;
  const className = "test-class";
  shallow(
    <ErrorBoundaryContainer
      {...params}
      className={className}
      view={props => (viewProps = props)}
    />
  );
  expect(viewProps.className).toEqual(className);
});

it("passes data-foo through to the view", () => {
  let viewProps;
  const data = "bar";
  shallow(
    <ErrorBoundaryContainer
      {...params}
      data-foo={data}
      view={props => (viewProps = props)}
    />
  );
  expect(viewProps["data-foo"]).toEqual(data);
});
