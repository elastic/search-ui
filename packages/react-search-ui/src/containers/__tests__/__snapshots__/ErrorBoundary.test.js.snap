// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supports a render prop 1`] = `
<div>
  I am an error
</div>
`;
